### ISSUE TITLE
using recent Pretrained models

### ISSUE THREAD

**Issue by smajida** [GeorgeSeif/Semantic-Segmentation-Suite#71]
This is a 
       1. Feature requests

Considering that ResNet has been around for a while and other recent networks have achieved better results compared with ResNet such as InceptionV3/4, ResNext, NASNet, PNASNet, and Xception, is there any plan to add these models along with ResNets?

According to https://github.com/tensorflow/models/tree/master/research/slim#Pretrained

Model                             | Top-1 Accuracy | Top-5 Accuracy
ResNet V2 152               | 77.8                   | 94.1
Inception V3                   | 78.0                   | 93.9
Inception V4                   | 80.2                   | 95.2
Inception-ResNet-v2      | 80.4                   | 95.3
NASNet-A_Large_331   | 82.7                   | 96.2
PNASNet-5_Large_331 | 82.9                   | 96.2
ResNext ?
Xception ?

Or at least some hints are appreciated to help me to add them and contribute to this repo by pulling this feature.

**Comment by GeorgeSeif** [comment]
Hmm sounds like a good idea!

To do so you need a few things:

- Download network weights
- Function definitions of the network structure, which returns end points at very key parts of the network. For example, with the current usage of ResNet, we return the block outputs at certain pooling layers.
- Add an option to use those networks in the files where the segmentation models are defined. 

I got the ResNet V2 networks (code and models) from [here](https://github.com/tensorflow/models/tree/master/research/slim).

Give it a shot if you'd like :). Once I finally sort things out and get my computer (hopefully within a month) I'll be adding a lot more to this repository!

**Comment by smajida** [comment]
then I will start with Inception V3.

**Comment by GeorgeSeif** [comment]
Awesome, thanks!

**Comment by GeorgeSeif** [comment]
Just to let you know, I recently added in MobileNetV2! Will be adding Dilated ResNet and NASNet-A_Large_331 soon!

**Comment by GeorgeSeif** [comment]
Okay, so I added InceptionV4 as well! Dilated ResNet I may or may not add soon. It will likely require some time due to many hand written code mods. 

Closing this issue for now!
